Generals/Code/GameEngine/Source/Common/RTS/Energy.cpp

```diff
@@ -188,7 +188,13 @@ void Energy::removePowerBonus( Object *obj )
 	if( obj == NULL )
 		return;
 
+	// TheSuperHackers @bugfix Caball009 14/11/2025 Don't remove power bonus for disabled power plants.
+#if RETAIL_COMPATIBLE_CRC
 	addProduction( -obj->getTemplate()->getEnergyBonus() );
+#else
+	if ( !obj->isDisabled() )
```

Comment:
I think this fix is not robust. It will break again when external power conditions are changed.

Either unify the exclusion condition into a static function and call this everywhere applicable, or give the modules a state that tells whether they gave energy to the player.

Reply:
Perhaps the `getObject()->isDisabled()` conditions simply need moving into both `addPowerBonus` and `removePowerBonus`. But yes can be follow up then.